packages/viewer/src/components/viewer-transform-widget/viewer-transform-widget.tsx

```diff
@@ -0,0 +1,367 @@
+import { Component, h, Host, Prop, Watch } from '@stencil/core';
+import { Point, Vector3 } from '@vertexvis/geometry';
+import { Color, Disposable } from '@vertexvis/utils';
+import classNames from 'classnames';
+
+import { readDOM, writeDOM } from '../../lib/stencil';
+import { TransformController } from '../../lib/transforms/controller';
+import { Mesh } from '../../lib/transforms/mesh';
+import {
+  computeUpdatedPosition,
+  convertCanvasPointToWorld,
+  convertPointToCanvas,
+} from './util';
+import { TransformWidget } from './widget';
+
+@Component({
+  tag: 'vertex-viewer-transform-widget',
+  styleUrl: 'viewer-transform-widget.css',
+  shadow: true,
+})
+export class ViewerTransformWidget {
+  /**
+   * The viewer to connect to measurements. If nested within a <vertex-viewer>,
+   * this property will be populated automatically.
+   */
+  @Prop()
+  public viewer?: HTMLVertexViewerElement;
+
+  /**
+   * The starting position of this transform widget. This position will be updated
+   * as transforms occur. Setting this value to `undefined` will remove the widget.
+   */
+  @Prop({ mutable: true })
+  public position?: Vector3.Vector3;
+
+  /**
+   * The controller that is responsible for performing transforms.
+   */
+  @Prop({ mutable: true })
+  public controller?: TransformController;
+
+  /**
+   * The color of the translation arrow on the x-axis. Defaults to `#ea3324`.
+   */
+  @Prop()
+  public xArrowColor: Color.Color | string = '#ea3324';
```

Comment:
Yeah, I had considered using CSS variables on the widget and parsing those within the component. I'll take a look at that since we have some prior work that has done similar, seems like it'd be reasonable here as well